### PR TITLE
Finish versioning of Staged_ledger_aux in RPC types

### PR DESCRIPTION
**Pull request by psteckler**
Within `Rpcs` functor, choose a version of `Staged_ledger_aux`. There's a comment to explain why that's justifiable.

Finish version-izing types used by that type.

- [ ] Tests were added for the new behavior
- [ ] All tests pass (CI will check this if you didn't)
- [X] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: src/lib/parallel_scan/parallel_scan.mli [hunk 1]
```diff
@@ -69,7 +69,19 @@ module State : sig
...
+
+  (* bin_io omitted intentionally *)
+  type ('a, 'd) t [@@deriving sexp]
+
+  module Stable : sig
+    module V1 :
+      sig
+        type ('a, 'd) t [@@deriving sexp, bin_io]
+      end
+      with type ('a, 'd) t = ('a, 'd) t
```
Comment (bkase) [inline src/lib/parallel_scan/parallel_scan.mli]: ocaml is crazy

File: src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml [hunk 2]
```diff
@@ -192,34 +213,13 @@ end = struct
...
 
-  include Binable.Of_binable
-            (T)
-            (struct
-              type nonrec t = t
-
-              let to_binable = Fn.id
-
-              let of_binable t =
-                assert (is_valid t) ;
```
Comment (bkase) [inline src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml]: did this assertion make it to your new code?

File: src/lib/parallel_scan/parallel_scan.mli (hunk 1 above)
Comment (psteckler) [reply src/lib/parallel_scan/parallel_scan.mli]: I'll move that equality to the enclosing signature, which looks less crazy.

File: src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml (hunk 2 above)
Comment (psteckler) [reply src/lib/transaction_snark_scan_state/transaction_snark_scan_state.ml]: Now it has.

**Pull request merged**